server/src/test/java/org/elasticsearch/index/shard/ShardGetServiceTests.java

```diff
@@ -114,6 +115,20 @@ public void testGetFromTranslogWithSyntheticSource() throws IOException {
         runGetFromTranslogWithOptions(docToIndex, sourceOptions, expectedFetchedSource, "\"long\"", 7L, true);
     }
 
+    public void testGetFromTranslogWithDenseVector() throws IOException {
+        float[] vector = new float[2048];
+        for (int i = 0; i < vector.length; i++) {
+            vector[i] = randomFloat();
+        }
+        String docToIndex = Strings.format("""
+            {
+                "bar": %s,
+                "foo": "foo"
+            }
+            """, Arrays.toString(vector));
+        runGetFromTranslogWithOptions(docToIndex, "\"enabled\": true", docToIndex, "\"text\"", "foo", "\"dense_vector\"", false);
```

Comment:
It would be nice to also validate that we can read out the right dense vector value, though with the way it is implemented, it should hardly be an issue.

Reply:
The test already checks that the retrieved source is identical: [source](https://github.com/elastic/elasticsearch/blob/d0263411cee3f258c4de7b5437c6b5149d24910b/server/src/test/java/org/elasticsearch/index/shard/ShardGetServiceTests.java#L187)
Do you think it's not enough?